flink-core/src/test/java/org/apache/flink/core/memory/HybridOffHeapDirectMemorySegmentTest.java

```diff
@@ -51,7 +51,7 @@ MemorySegment createSegment(int size, Object owner) {
     @Test
     public void testHybridHeapSegmentSpecifics() {
         final int bufSize = 411;
-        HybridMemorySegment seg = (HybridMemorySegment) createSegment(bufSize);
+        OffHeapMemorySegment seg = (OffHeapMemorySegment) createSegment(bufSize);
```

Comment:
The cast should not be necessary if we change the return type of `createSegment`.